FlorianGrollich/Banana
# Feature/login backend integration

### Summary

This PR adds backend enhancements for login and registration functionality and TypeScript configuration. Environmental variables are added and dependencies are updated.

### Key Changes

- Login & registration backend integration.
- Integrated dotenv for environment variable management.
- Updated package.json to include a build script for TypeScript compilation.
- Modified tsconfig.json to specify the output directory and improve module resolution.
- Refactored data-source.ts to use ES6 module imports and exports.
- Cleaned up routes.ts by removing unused imports.

### Testing

- Ensure there are no build errors in the server.
- Server runs on localhost:3200.
- Verify database connections and route handlers are functioning correctly.

### Review Requests

- Please check that there are no build and type errors.
- Please check the dependency setup and the TypeScript configurations.
- Ensure that the .env file is properly set up according to the new dotenv integration.

Thank you!
